packages/terra-dialog/src/Dialog.jsx

```diff
@@ -3,24 +3,29 @@ import PropTypes from 'prop-types';
 import classNames from 'classnames/bind';
 import Button from 'terra-button';
 import ContentContainer from 'terra-content-container';
+import { injectIntl, intlShape } from 'react-intl';
 import styles from './Dialog.module.scss';
 
 const cx = classNames.bind(styles);
 
 const propTypes = {
   /**
-   * React node to be placed within the header area of the dialog.
+   * The children to be placed within the main content area of the dialog.
    */
-  header: PropTypes.node.isRequired,
+  children: PropTypes.node,
   /**
    * React node to be placed within the footer area of the dialog.
    * Content exceeding the available space will be hidden.
    */
   footer: PropTypes.node.isRequired,
   /**
-   * The children to be placed within the main content area of the dialog.
+   * React node to be placed within the header area of the dialog.
    */
-  children: PropTypes.node,
+  header: PropTypes.node.isRequired,
+  /**
+   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
```

Comment:
We should add the `@private` syntax so this doesn't show up in the generated prop-types documentation.